lib/src/utils.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Creates a human readable text from camel case
+ * @param {string} value the camel case string (example "caseId")
+ * @param {boolean} shouldCapitalizeEvery OPTIONAL should every subsequest word be capitalized? Defaults to false
+ * @returns the a capitalised human readable string (example "Case Id" or "Case id")
+ */
+export function readableFromCamelCase(value: string, shouldCapitalizeEvery?: boolean): string {
```

Comment:
You Are Making Complex A Very Easy Thing. You Can Add This In Your CSS.
each table header should be able to accept a className to apply whatever styles individually.
If you want to capitalize then:
`text-transform: capitalize; `
you can also try providing the text that you want to appear directly to the header by modifying the type of each header and adding the property text to it

Reply:
your suggestion would display this
![image](https://user-images.githubusercontent.com/49055952/84118773-07a71d00-aa66-11ea-8578-40dbb71a0b34.png)

instead of 
![image](https://user-images.githubusercontent.com/49055952/84118855-27d6dc00-aa66-11ea-92cf-3ec72c535f49.png)

I there any text-transform feature which would be able to display camelCase as human readable?

Reply:
i cant remember if there is any, but i doubt it.
The idea here is to avoid writing a script to do this for you instead you should be able to populate the header of the table with values that you choose, here is an example:

The label is the value that will be shown to the user, in this example its a key because each header cell have a translate pipe that will grab the key and translate it to the correct value.

you are not using the translate pipe inside the table component, but atleast you will give a choice for who wants to use it to put the correct labels and usually those ones are already well formated in the CMS without the need to transform them using javascript,

```typescript
[
      {
        label: 'label_typeofsaving',
        classNames: ['col-3', 'sort'],
        tableKeySelector: 'policyTypeName',
        active: true,
        asc: true,
        valueType: 'string'
      },
      {
        label: 'label_owner',
        classNames: ['col-2', 'sort'],
        tableKeySelector: 'policyHolder',
        active: null,
        asc: null,
        valueType: 'string'
      },
      {
        label: 'text_investmentadvice',
        classNames: ['col-2', 'sort'],
        tableKeySelector: 'adviceStatus',
        active: null,
        asc: null,
        valueType: 'number'
      },
      {
        label: 'label_value',
        classNames: ['col-2', 'text-right', 'sort'],
        tableKeySelector: 'insuranceAmount',
        active: null,
        asc: null,
        valueType: 'number'
      },
      {
        label: 'label_totaldevelopment',
        classNames: ['col-2', 'text-right', 'sort', 'total-development'],
        tableKeySelector: 'totalDevelopment',
        active: null,
        asc: null,
        valueType: 'number'
      },
      {
        label: '',
        classNames: ['col-1'],
        tableKeySelector: '',
        active: null,
        asc: null,
        valueType: null
      }
    ]
```

Reply:
There is already an api provided for the user to put custom labels using the tableConfig
Here is the TS definition for it 
https://github.com/sebgroup/ng-components/pull/293/files#diff-6249b4175d128008cd94ca578d959de3R15

It will use the `readableFromCamelCase` by default if you do not provide (override) the label.

Reply:
this is the logic where the table header is constructed and the label used if provided: 
https://github.com/sebgroup/ng-components/pull/293/files#diff-7540fde1b76eeede1a2842a0a140ffcbR231

Reply:
So since this transformation is only happening at initialization, once for every column name I don't see this as a performance issue so I will keep it for for now.